### DISCUSSION TITLE
Have USB cam installed and working. Can we get a time lapse feature installed? if so how?

### DISCUSSION THREAD

**Discussion by DougJoseph** [xblax/flashforge_ad5m_klipper_mod#84] [Q&A] [answered]
I tried to follow the steps for Moonraker Timelapse: https://github.com/mainsail-crew/moonraker-timelapse/blob/main/docs/installation.md

However, after I SSH in and get to the command to 

`git clone http://github.com/mainsail-crew/moonraker-timelapse.git
`
I get this:

```
Cloning into 'moonraker-timelapse'...
fatal: unable to access 'https://github.com/mainsail-crew/moonraker-timelapse.git/': SSL certificate problem: unable to get local issuer certificate
```

And when I tried to leave off the "S" in https, I got this:

```
fatal: unable to update url base from redirection:
  asked for: http://github.com/mainsail-crew/moonraker-timelapse.git/info/refs?service=git-upload-pack
   redirect: http://filter.meetcircle.com/prek/?filtered=github.com&cat=Science+&amp;+Technology&catid=25&reason=blocked
```

This appreciative Klipper newbie welcomes any and all help.

Thanks again for this mod!

**Comment by DougJoseph** [comment]
Hey, I think I know what's wrong with the "Redirect" issue.

I noticed this tidbit in the redirect wording: "redirect: http://filter.**meetcircle.com**

...And that means my home internet filter, which is the "Circle" brand of family filter protection, is blocking the printer from getting access to the github site.

I will now work to correct that. 

**UPDATE:**

That was indeed why the command to clone the repo was not working! After I got it authorized to get past the filter, it cloned the repo.

However, now while I can CD into the "moonraker-timelapse" directory, when I try the last command, "make install," I get 

`-sh: make: not found
`

Here are the commands in the readme:

```
cd ~/
git clone https://github.com/mainsail-crew/moonraker-timelapse.git
cd ~/moonraker-timelapse
make install
```

**Reply by MACCAMENZIES** [reply]
In Linux
sudo apt-get install make

**Reply by KaruroChori** [reply]
On debian based systems. This is not no debian based system, so no apt for us.

**Reply by MACCAMENZIES** [reply]
Apologies.  What linux base is it ?

**Reply by xblax** [reply]
Embedded Linux based on Buildroot, so no packet manager. But @consp has made great progress regarding the timelapse plugin, maybe we will have it soon! 
https://github.com/xblax/flashforge_adm5_klipper_mod/issues/119#issuecomment-2041536593

**Comment by consp** [comment]
Unfortunately it won't work even if you get it to run.

Moonraker timelapse (and all other timelaps encoders) use ffmpeg. ffmpeg is not included as it uses way too much memory. See #65 #74 . It will cause everything to stop, wait for the kernel to kill the offending process and in the meantime your print will be in ruins.

I've look in using the build in hardware acceleration available mainly to speed up the screen and that includes the hardware encoder. Unfortunately it's not trivial to use and this is definitely a long term project, expect no results.

The best way to get timelapses is to encode them somewhere else. The videostream _is_ hardware accelerated and can be used outside of the printer.


For future viewers: 
It's not possible to compile anything on the printer and thus make is not installed. It's _VERY_ limited. It can run moonraker/fluidd/klipperscreen but that's about it. Anything requiring installing stuff (e.g. compiling something, building files etc) will not work and must be done during the build process of the installation (as in on a different machine).

Expect any "install.sh" script to not work. It's generally expected to be a very modern raspberry pi or full fledged computer which are an order of mangnitude faster and have an order of magnitude more memory than the processor on the printer.

**Reply by consp** [reply]
> Moonraker timelapse (and all other timelaps encoders) use ffmpeg. ffmpeg is not included as it uses way too much memory. See https://github.com/xblax/flashforge_adm5_klipper_mod/issues/65 https://github.com/xblax/flashforge_adm5_klipper_mod/discussions/74 . It will cause everything to stop, wait for the kernel to kill the offending process and in the meantime your print will be in ruins

I'll be actually trying to get something working. Maybe it will work in the end (no promisses).

**Comment by DougJoseph** [comment]
OK, got it! Thanks! I understand.

**Reply by consp** [reply]
Maybe if you get something working off-printer you can include a doc/guide somewhere? Would be a nice to have feature.

**Comment by DougJoseph** [comment]
So, is there news on this? I noticed the main page / read me now says there is support for Moonraker Timelapse. Did someone find a way to install it? Here's where I hit a snag last time.

> When I try the last command, "make install," I get
> 
> -sh: make: not found 
> 
> Here are the commands in the readme:
> ```
> cd ~/
> git clone https://github.com/mainsail-crew/moonraker-timelapse.git
> cd ~/moonraker-timelapse
> make install
> ```

...Or maybe the current build of the mod here, already includes what is needed?

Thanks for any help!

**Comment by DougJoseph** [comment, marked as answer]
OK, so I see now that the latest version (Adventurer5M-KlipperMod-v00.05-beta) has all the time lapse stuff already baked in. Thank you! Y'all are awesome.
